server/src/server.js

```diff
@@ -1,337 +1,189 @@
-      data.backend = backend;
-
-      let response = React.renderToString(<Handler {...data} />),
-        footerIndex = response.indexOf(footer),
-        header = response.slice(0, footerIndex);
-
-      res.setHeader(contentType, "text/html; charset=utf-8");
-      res.end(
-        doctype +
-        header +
-        "<script>var cachedData = " + JSON.stringify(data) + ";" +
-        script +
-        "</script>" +
-        footer);
-    }).catch(function (e) {
-      console.log("Error:", e.stack);
-      res.setHeader(contentType, "text/plain; charset=utf-8");
-      res.writeHead(500);
-      if (e.code == "ECONNREFUSED") {
-        // Database connection error
-        res.end("Cannot connect to database");
-      } else {
-        res.end(e.stack);
-      }
-    });
+    return null;
+  }).then(() => {
+    simpleResponse(res, "Saved", 200);
+  }).catch((err) => {
+    errorResponse(res, "Error saving Object:", err);
   });
 });
 
-function handleRegister(req, res) {
-  parsePost(req).then(function (vars) {
-    // FIXME: should also handle updating
-    // FIXME: need to hash secret
-    let canUpdate = vars.userId === req.userId;
-    return registerLogin(vars.userId, {
-      secret: vars.secret,
-      nickname: vars.nickname || null,
-      avatarurl: vars.avatarurl || null
-    }, canUpdate).then(function (ok) {
-      if (ok) {
-        let cookies = new Cookies(req, res, dbschema.getKeygrip());
-        cookies.set("user", vars.userId, {signed: true});
-        simpleResponse(res, "Created", 200);
-      } else {
-        simpleResponse(res, "User exists", 401);
+app.get("/data/:id/:domain", function (req, res) {
+  let shotId = req.params.id + "/" + req.params.domain;
+  Shot.getRawValue(shotId).then((data) => {
+    if (! data) {
+      simpleResponse(res, "No such shot", 404);
+    } else {
+      let value = data.value;
+      if ('format' in req.query) {
+        value = JSON.stringify(JSON.parse(value), null, '  ');
       }
-    });
+      res.setHeader(contentType, "application/json");
+      res.end(value);
+    }
   }).catch(function (err) {
-    errorResponse(res, "Error registering:", err);
+    errorResponse(res, "Error serving data:", err);
   });
-}
+});
 
-function parsePost(req) {
-  return new Promise((resolve, reject) => {
-    let body = '';
-    req.on('data', function (data) {
-      body += data;
-      // Too much POST data, kill the connection!
-      if (body.length > 1e6) {
-        req.connection.destroy();
-        reject(new Error("Request body too large"));
-      }
-    });
-    req.on('end', function () {
-      let post = qs.parse(body);
-      resolve(post);
-    });
+app.get("/content/:id/:domain", function (req, res) {
+  let shotId = req.params.id + "/" + req.params.domain;
+  Shot.get(req.backend, shotId).then((shot) => {
+    if (! shot) {
+      simpleResponse(res, "Not found", 404);
+      return;
+    }
+    res.writeHead(200);
+    res.end(shot.staticHtml({
```

Comment:
Leftover stuff from before Express – but probably I should search for all .end()s and update them